quickwit-indexing/src/source/kafka_source.rs

```diff
@@ -268,13 +277,12 @@ impl Source for KafkaSource {
         batch_sink: &Mailbox<Indexer>,
         ctx: &SourceContext,
     ) -> Result<Duration, ActorExitStatus> {
+        let mut batch_num_bytes = 0;
         let mut docs = Vec::new();
         let mut checkpoint_delta = CheckpointDelta::default();
 
-        let deadline = tokio::time::sleep(quickwit_actors::HEARTBEAT / 2);
-        let mut message_stream = Box::pin(self.consumer.stream().take_until(deadline));
```

Comment:
Removing the stream since we can directly poll the deadline future with `tokio::select!`.